plugins/scaffolder-backend/api-report.md

```diff
@@ -44,6 +45,10 @@ export type ActionContext<Input extends JsonObject> = {
   createTemporaryDirectory(): Promise<string>;
   templateInfo?: TemplateInfo;
   isDryRun?: boolean;
+  user?: {
+    entity?: UserEntity;
```

Comment:
To not create a breaking changes, we can provide an email instead.
But when `UserEntity` going to be changed, anyway all application has to be adopted (up to you to decide).

But loading from the action, currently is not possible, as there is no correlation atm how you can deduct the user. 
Otherwise I'd just do it, without any backstage modifications :)

Moreover if to look at other frameworks, usually you have an information about author/creator/modifier. 
We can change the type, if it will help to solve the concern with future evolving of backstage ;)

Reply:
My suggestion would be to just pass in the user entity ref (the field below) but not resolving it. But I'm deferring to @benjdlambert to decide

Reply:
It's fine by me.
Only in this case we would need also to inject `CatalogApi`

Reply:
Hmm well actually if it's just the user entity that you want to pass you can already do that from the template itself.

You can do something like this:

```yaml
action: ..
input:
  userRef: ${{ user.ref }}
```

Theres also `${{ user.entity }}` which is the actual entity, but the problem with using this would be that you would have to type the input to the action with the entire entity, which could be for better or for worse.

Wondering if actually like @freben said it would be more beneficial to just use the ref, and you can check the `apiVersion` etc in the action when using the `catalog` client.

Reply:
@benjdlambert Technically you right. it's also possible to do like that. Only it will create a security vulnerability. 

Imagine that from backstage you allow to read from this template, or some project where all templates are located and user is passed this way. This input of `${{ user.entity }}` or `userRef: ${{ user.ref }}` can be changed to some hard coded name to escalate permissions. 

There is way more secure to do it inside of backstage plugin, which will inject it from identity provider. 
Giving a possibility for the end user to inject the user is not trustworthy.

Reply:
Yeah, that's a good point if you're treating the template as trusted. 

Ok, I think this alone is a good enough reason that we should provide some way to grab this in actions instead.